pkg/k8s/node.go

```diff
@@ -230,15 +248,67 @@ func setNodeNetworkUnavailableFalse(c kubernetes.Interface, nodeName string) err
 	return err
 }
 
+// removeNodeTaint removes the AgentNotReadyNodeTaint allowing for pods to be
+// scheduled once Cilium is setup. Mostly used in cloud providers to prevent
+// existing CNI plugins from managing pods.
+func removeNodeTaint(ctx context.Context, c kubernetes.Interface, nodeGetter nodeGetter, nodeName string) error {
+	k8sNode, err := nodeGetter.GetK8sNode(ctx, nodeName)
+	if err != nil {
+		return err
+	}
+	i := 0
+	var taintFound bool
+	for _, taint := range k8sNode.Spec.Taints {
+		if taint.Key != ciliumio.AgentNotReadyNodeTaint {
+			k8sNode.Spec.Taints[i] = taint
+			i++
+		} else {
+			taintFound = true
+		}
+	}
```

Comment:
This code is kinda strange because we're iterating over a slice that we modify. Why not do something like:
```go
taints := make([]..., 0, len(k8sNode.Spec.Taints)
for _, taint := range k8sNode.Spec.Taints {
	if taint.Key != ciliumio.AgentNotReadyNodeTaint {
		taints = append(taints, taint)
	} else {
		taintFound = true
	}
}
```

Then later
```go
k8sNode.Spec.Taints = taints
```

Yes it will make a copy, but this is much clearer and the taints aren't large objects anyway so the performance hit will probably be negligible.

Reply:
Oh good point... I'm not sure why I went with my approach.